platforms/fabric/src/main/java/com/dfsek/terra/fabric/generation/FabricChunkGeneratorWrapper.java

```diff
@@ -105,52 +113,15 @@ public void buildSurface(ChunkRegion region, StructureAccessor structures, Chunk
     @Override
     public void populateEntities(ChunkRegion region) {
         ChunkPos chunkPos = region.getCenterPos();
-        Biome biome = region.getBiome(chunkPos.getStartPos().withY(region.getTopY() - 1));
+        RegistryEntry<Biome> biome = region.getBiome(chunkPos.getStartPos().withY(region.getTopY() - 1));
         ChunkRandom chunkRandom = new ChunkRandom(new AtomicSimpleRandom(RandomSeed.getSeed()));
         chunkRandom.setPopulationSeed(region.getSeed(), chunkPos.getStartX(), chunkPos.getStartZ());
         SpawnHelper.populateEntities(region, biome, chunkPos, chunkRandom);
-        
     }
     
     @Override
     public int getWorldHeight() {
-        return settingsSupplier.get().getGenerationShapeConfig().height();
-    }
-    
-    public Pool<SpawnSettings.SpawnEntry> getEntitySpawnList(Biome biome, StructureAccessor accessor, SpawnGroup group, BlockPos pos) {
-        if(!accessor.hasStructureReferences(pos)) {
-            return super.getEntitySpawnList(biome, accessor, group, pos);
-        } else {
-            if(accessor.getStructureAt(pos, StructureFeature.SWAMP_HUT).hasChildren()) {
-                if(group == SpawnGroup.MONSTER) {
-                    return SwampHutFeature.MONSTER_SPAWNS;
-                }
-                
-                if(group == SpawnGroup.CREATURE) {
-                    return SwampHutFeature.CREATURE_SPAWNS;
-                }
-            }
-            
-            if(group == SpawnGroup.MONSTER) {
-                if(accessor.getStructureAt(pos, StructureFeature.PILLAGER_OUTPOST).hasChildren()) {
-                    return PillagerOutpostFeature.MONSTER_SPAWNS;
-                }
-                
-                if(accessor.getStructureAt(pos, StructureFeature.MONUMENT).hasChildren()) {
-                    return OceanMonumentFeature.MONSTER_SPAWNS;
-                }
-                
-                if(accessor.getStructureAt(pos, StructureFeature.FORTRESS).hasChildren()) {
-                    return NetherFortressFeature.MONSTER_SPAWNS;
-                }
-            }
-            
-            return (group == SpawnGroup.UNDERGROUND_WATER_CREATURE || group == SpawnGroup.AXOLOTLS) && accessor.getStructureAt(pos,
-                                                                                                                               StructureFeature.MONUMENT)
-                                                                                                               .hasChildren()
-                   ? SpawnSettings.EMPTY_ENTRY_POOL
-                   : super.getEntitySpawnList(biome, accessor, group, pos);
-        }
```

Comment:
Was this reintroduced somewhere else?

Reply:
it got magicked away somewhere in Minecraft code (no longer decided by the chunk generator), still have to test whether it works with our structure injection.